jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java

```diff
@@ -869,12 +869,44 @@ public List<String> getSoftDeleteProperties(Class<?> clazz) {
      * @return true if entity's {@link EntityEntry} implements {@link EntityEntryHasUuid}
      */
     public boolean hasUuid(MetaClass metaClass) {
-        return metaClass.getAnnotations().containsKey(UUID_KEY_ANN_NAME);
+        String uuidPropertyName = getUuidPropertyName(metaClass);
+        return uuidPropertyName != null;
     }
 
+    /**
+     * Gets name of uuid key property.
+     *
+     * @param clazz entity java class
+     * @return uuid property name or null if it doesn't exist
```

Comment:
After your changes the method will throw an exception if there is no metaClass for the given java class. It is worth mentioning in the javadoc.

Reply:
Done